src/lib/cmp.js

```diff
@@ -84,16 +85,23 @@ export default class Cmp {
 		 * @param {Array} vendorIds Array of vendor IDs to retrieve.  If empty return all vendors.
 		 */
 		getVendorConsents: (vendorIds, callback = () => {}) => {
+			const { purposeConsents, vendorConsents } = this.store.getVendorConsentsObject(vendorIds);
 			const consent = {
-				metadata: this.generateConsentString(),
+				metadata: this.generateMetadataString(),
 				gdprApplies: this.gdprApplies,
 				hasGlobalScope: this.config.storeConsentGlobally,
-				...this.store.getVendorConsentsObject(vendorIds)
+				purposeConsents,
+				vendorConsents
 			};
 
 			callback(consent, true);
 		},
 
+		decodeMetadata: (_, callback = () => {}) => {
```

Comment:
What's the purpose of having an extra argument here? It could be at least optional.

Reply:
This is to fit with the format of the other CMP commands (like `ping` and `getVendorConsents`). The three arguments to any `window.__cmp()` function call are (function, params, callback). In this case, the params are not used, so we discard them.

Reply:
Made behaviour same as in `ping` command.